### PR TITLE
fix(AudioV2): resume looping static sound from correct position with loopStart/loopEnd

### PR DESCRIPTION
**Pull request by RaananW**
## Problem

When an AudioV2 static sound with a loop sub-range (`loopStart` / `loopEnd`) is paused and then resumed, playback always restarts at `loopStart` instead of continuing from where it was paused.

Reported on the forum: [Audio v2 resume audio sprites not working](https://forum.babylonjs.com/t/audio-v2-resume-audio-sprites-not-working/63743). This also affects the "audio sprite" use case (multiple sounds sharing one buffer, each with its own `startOffset` / `loopStart` / `loopEnd`).

Repro playground: https://playground.babylonjs.com/#6FZCRD#1

## Root cause

In `_WebAudioStaticSoundInstance.play()`, the resume offset was computed as:

```ts
startOffset += this._enginePauseTime;
startOffset %= this._sound.buffer.duration;
```

This wraps against the whole buffer duration and ignores the loop region. When `loopStart`/`loopEnd` describe a sub-range, the computed offset can fall outside `[loopStart, loopEnd)`, so the underlying `AudioBufferSourceNode` (which is looping) restarts playback at `loopStart`.

## Fix

When resuming a looping sound, map the elapsed playback time onto the actual loop region: play the initial `startOffset → loopEnd` segment, then wrap the remaining time within `[loopStart, loopEnd)`. Non-looping behavior is unchanged.

The loop bounds are validated the same way Web Audio treats them (falling back to the full buffer when `loopStart`/`loopEnd` don't describe a valid sub-range).

## Testing

- Added regression test `Play looping sound with loopStart and loopEnd options set, pause it, and resume it` to `staticSound.playback.test.ts`.
- Verified the new test **passes with** the fix and **fails without** it (produces `[1, 2, 1, 1, 2]` instead of the expected `[1, 2, 1, 2, 1]`), so it genuinely guards the regression.
- All 43 existing static-sound playback + currentTime Playwright tests pass, including existing pause/resume and loop cases.

Note: streaming sounds use a separate backend and are out of scope here; buffer-backed sounds go through this same static-sound instance path and are covered.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/18661/merge

Test environment:
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18661/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18661/merge/index.html#WGZLGJ#4600

Links to test your changes to core in the published versions of the Babylon tools (does not contain changes you made to the tools themselves):

https://playground.babylonjs.com/?snapshot=refs/pull/18661/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/18661/merge
https://gui.babylonjs.com/?snapshot=refs/pull/18661/merge
https://nme.babylonjs.com/?snapshot=refs/pull/18661/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/18661/merge#BCU1XR#0

*If you made changes to the sandbox or playground in this PR, additional comments will be generated soon containing links to the dev versions of those tools.*

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18661/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18661/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
## ⚡ Performance Test Results

🟢 All performance tests passed — no regressions detected.

**Comment by bjsplat** [comment]
Interaction tests

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18661/merge/testResults/interactionplaywright/index.html

**Comment by bjsplat** [comment]
## 🟢 Memory Leak Test Results

**4** passed, **0** leaked out of **4** scenarios

🟢 All memory leak tests passed — no leaks detected.

<details>
<summary>Passed Scenarios (4)</summary>

| Scenario | Package |
|----------|---------|
| Core Playground #2FDQT5#1508 | @babylonjs/core |
| Core Playground #T90MQ4#14 | @babylonjs/core |
| Core Playground #8EDB5N#2 | @babylonjs/core |
| Core Playground #LL5BIQ#636 | @babylonjs/core |

</details>

**Review by ryantrem — APPROVED** [review]

**Comment by bjsplat** [comment]
Audio test results:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18661/merge/testResults/audioV2/index.html

File: packages/dev/core/src/AudioV2/webAudio/webAudioStaticSound.ts [hunk 1]
```diff
@@ -409,8 +409,27 @@ class _WebAudioStaticSoundInstance extends _StaticSoundInstance implements IWebA
...
-            startOffset %= this._sound.buffer.duration;
+            const duration = this._sound.buffer.duration;
+            const elapsed = this._enginePauseTime;
+
+            if (this._options.loop) {
+                // Web Audio loops over the whole buffer when `loopStart`/`loopEnd` do not describe a valid sub-range,
+                // so mirror that here to compute the correct resume position inside the loop region.
+                const loopEnd = this._options.loopEnd > 0 && this._options.loopEnd <= duration ? this._options.loopEnd : duration;
+                const loopStart = this._options.loopStart >= 0 && this._options.loopStart < loopEnd ? this._options.loopStart : 0;
+
```
[the remark this answers is not in the document]
Comment (RaananW) [reply packages/dev/core/src/AudioV2/webAudio/webAudioStaticSound.ts]: Fixed ✅ — The resume loop bounds now follow the exact \`AudioBufferSourceNode\` playback algorithm: use \`[loopStart, min(loopEnd, duration))\` only when \`loopStart >= 0 && loopEnd > 0 && loopStart < loopEnd\`, otherwise fall back to the whole buffer \`[0, duration)\`. This removes the per-variable fallback that could turn an invalid range (e.g. \`loopStart >= loopEnd\`) into a different short valid sub-range. I verified the new normalization matches the spec across all 49 combinations of edge-case \`loopStart\`/\`loopEnd\` inputs (0 mismatches vs 24 for the previous logic).

**Comment by bjsplat** [comment]
## ⚡ Performance Test Results

🟢 All performance tests passed — no regressions detected.

**Comment by bjsplat** [comment]
Visualization tests for WebGPU

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18661/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
Interaction tests

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18661/merge/testResults/interactionplaywright/index.html

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18661/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
## 🟢 Memory Leak Test Results

**4** passed, **0** leaked out of **4** scenarios

🟢 All memory leak tests passed — no leaks detected.

<details>
<summary>Passed Scenarios (4)</summary>

| Scenario | Package |
|----------|---------|
| Core Playground #2FDQT5#1508 | @babylonjs/core |
| Core Playground #T90MQ4#14 | @babylonjs/core |
| Core Playground #8EDB5N#2 | @babylonjs/core |
| Core Playground #LL5BIQ#636 | @babylonjs/core |

</details>

**Comment by bjsplat** [comment]
Audio test results:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18661/merge/testResults/audioV2/index.html

**Review by sebavan — APPROVED** [review]
